code/datastructure-set/src/main/java/ru/unn/agile/datastructure/set/model/MySet.java

```diff
@@ -0,0 +1,97 @@
+package ru.unn.agile.datastructure.set.model;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+
+public class MySet<E> {
+
+    private static final String ERROR_MESSAGE = "MySet collection doesn't support Null values";
+    private List<E> container;
+
+    public MySet() {
+        container = new ArrayList<>();
+    }
+
+    public int size() {
+        return this.container.size();
+    }
+
+    public boolean add(final E e) {
+        if (e == null) {
+            throw new NullPointerException(ERROR_MESSAGE);
+        }
+
+        if (!this.container.contains(e)) {
```

Comment:
You check, if container contains the element here, but you don't do it in addAll function. Please use one style

Reply:
because the addAll method rise the add function which performs contains validation.